pkg/cmd/adm/restart.go

```diff
@@ -3,155 +3,198 @@ package adm
-	}
-	term.PrintContextSeparatorWithBodyf(deploymentList, "Existing deployments in %s namespace", ns)
 	return nil
+
 }
 
-func scaleToZero(cl runtimeclient.Client, namespacedName types.NamespacedName) (int32, error) {
-	// get the deployment
-	deployment := &appsv1.Deployment{}
-	if err := cl.Get(context.TODO(), namespacedName, deployment); err != nil {
-		return 0, err
+func restartNonOlmDeployments(deployment appsv1.Deployment, f cmdutil.Factory, ioStreams genericclioptions.IOStreams) error {
+
+	o := kubectlrollout.NewRolloutRestartOptions(ioStreams)
+
+	if err := o.Complete(f, nil, []string{"deployment"}); err != nil {
+		panic(err)
 	}
-	// keep original number of replicas so we can bring it back
-	originalReplicas := *deployment.Spec.Replicas
-	zero := int32(0)
-	deployment.Spec.Replicas = &zero
 
-	// update the deployment so it scales to zero
-	return originalReplicas, cl.Update(context.TODO(), deployment)
+	o.Resources = []string{"deployment/" + deployment.Name}
+
+	if err := o.Validate(); err != nil {
+		panic(err)
+	}
+	fmt.Printf("Running the rollout restart command for non-olm deployment %v", deployment)
+	return o.RunRestart()
 }
 
-func scaleBack(term ioutils.Terminal, cl runtimeclient.Client, namespacedName types.NamespacedName, originalReplicas int32) error {
-	return wait.Poll(500*time.Millisecond, 10*time.Second, func() (done bool, err error) {
-		term.Println("")
-		term.Printlnf("Trying to scale the deployment back to '%d'", originalReplicas)
-		// get the updated
-		deployment := &appsv1.Deployment{}
-		if err := cl.Get(context.TODO(), namespacedName, deployment); err != nil {
-			return false, err
-		}
-		// check if the replicas number wasn't already reset by a controller
-		if *deployment.Spec.Replicas == originalReplicas {
-			return true, nil
-		}
-		// set the original
-		deployment.Spec.Replicas = &originalReplicas
-		// and update to scale back
-		if err := cl.Update(context.TODO(), deployment); err != nil {
-			term.Printlnf("error updating Deployment '%s': %s. Will retry again...", namespacedName.Name, err.Error())
-			return false, nil
-		}
-		return true, nil
-	})
+func checkRolloutStatus(f cmdutil.Factory, ioStreams genericclioptions.IOStreams, labelSelector string) error {
+	cmd := kubectlrollout.NewRolloutStatusOptions(ioStreams)
+
+	if err := cmd.Complete(f, []string{"deployment"}); err != nil {
+		panic(err)
+	}
+	cmd.LabelSelector = labelSelector
+	if err := cmd.Validate(); err != nil {
+		panic(err)
+	}
+	fmt.Printf("Running the Rollout status to check the status of the deployment")
+	return cmd.Run()
+}
+
+func getExistingDeployments(cl runtimeclient.Client, ns string) (*appsv1.DeploymentList, *appsv1.DeploymentList, error) {
+
+	olmDeployments := &appsv1.DeploymentList{}
+	if err := cl.List(context.TODO(), olmDeployments,
+		runtimeclient.InNamespace(ns),
+		runtimeclient.MatchingLabels{"kubesaw-control-plane": "kubesaw-controller-manager"}); err != nil {
```

Comment:
minor thing - I see that we wanted to rely on our own label, but in case we introduce a new olm based operator we need to remember to add this label and avoid adding it in case we introduce a non olm deployment. 

What was the problem with relying on the `olm.managed: true` label provided by OLM ?

Reply:
the label olm.managed: true is a generic label which can restart other olm deployments  and we want to avoid that scenario and hence our specific label

Reply:
I see, thanks for explaining this. 
I don't think we should have other operators deployed in the same namespace as our host and member operator, but maybe it is a use case 👍